### PR TITLE
Upgrade halide zynq runtime to halide_buffer_t

### PR DESCRIPTION
**Pull request by Kuree**
This PR, based on #28, upgraded `HalideRuntimeZynq` to use `halide_buffer_t` instead of deprecated `buffer_t`.

For applications relying on Zynq CodeGen, they need to link `HalideRuntimeZynq` to interact with the kernel drivers.

### CODE REVIEW

**Comment by jingpu** [comment]
Thanks for the fix! Can you add a minimal test for the Zynq build in the travis script, i.e. basically creating a make target and adding it to https://github.com/jingpu/Halide-HLS/blob/HLS/test/scripts/build_travis.sh. Let me know if you need a help.

**Comment by Kuree** [comment]
Done. Later this week I will add changes to the the CodeGen that also produces `pipeline_zynq` function that accepts `fd_hwacc` and `fd_cma`. f4graph needs that to be able to handle multiple cameras/hls, i.e. multiple `fd_hwacc`.

Tl;DR: more commits to this PR are coming.

File: Makefile [hunk 1]
```diff
@@ -697,6 +697,10 @@ ifeq ($(UNAME), Darwin)
 	install_name_tool -id $(CURDIR)/$(BIN_DIR)/libHalide.$(SHARED_EXT) $(BIN_DIR)/libHalide.$(SHARED_EXT)
 endif
 
+zynq_runtime: $(LIB_DIR)/libHalide.a $(ROOT_DIR)/apps/hls_examples/hls_support/HalideRuntimeZynq.cpp $(INCLUDE_DIR)/HalideRuntime.h
```
Comment (jingpu) [inline Makefile]: Do you need `$(LIB_DIR)/libHalide.a` in the dependency?

**Review by jingpu — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (Kuree) [reply Makefile]: Good catch. Actually it's not necessary. I will remove that.
